content/dart/concepts/type-conversion/type-conversion.md

```diff
@@ -0,0 +1,496 @@
+```Shell
+123000000000000000000
+1.23e+22
+```
+
+---
+
+## Variations Of `toString` Method
+
+* `toStringAsFixed()`[^6]
+* `toStringAsExponential()`[^7]
+* `toStringAsPrecision()`[^8]
+
+---
+
+## toStringAsFixed() Method
+
+## Implementation
+
+```pseudo
+String toStringAsFixed(int fractionDigits);
+```
+
+The `toStringAsFixed` method converts the input number to a `double` before computing the string representation of that number.
+
+The parameter `fractionDigits` must be an integer satisfying:
+
+  0 <= `fractionDigits` <= 20.
+
+Meaning the `fractionDigits` must be a Non-Zero and a Positive value and must be less than 21.
+
+If `fractionDigits` equals 0, then the decimal point is omitted.
+
+If the absolute value of the input is greater than or equal to 10^21, then this methods returns an exponential representation computed by `toStringAsExponential()`.
+
+## toStringAsFixed Method Example
+
+```dart
+  void main() {
+  var a = 1.toStringAsFixed(3);
+  var b = (256463.51554546).toStringAsFixed(3);
+  var c = (2351.44464661).toStringAsFixed(5);
+  var d = 254645466111.toStringAsFixed(3);
+  var e = 1224.toStringAsFixed(20);
+  var f = 5.25.toStringAsFixed(0);
+  
+  print(a);
+  print(b);
+  print(c);
+  print(d);
+  print(e);
+  print(f);
+}
+```
+
+The `Shell` output:
+
+```Shell
+1.000
+256463.516
+2351.44465
+254645466111.000
+1224.00000000000000000000
+5
+```
+
+> **Note:** If the `fractionDigit` value is not in specified range, console output will display a `RangeError`.
+
+## Range Error Example
+
+```dart
+  var a = 1.toStringAsFixed(21);
+  
+  print(a);
+```
+
+The `Shell` output:
+
+```Shell
+Uncaught Error: RangeError (fractionDigits): Invalid value: Not in inclusive range 0..20: 21
+```
+
+---
+
+## toStringAsExponential() Method
+
+## toStringAsExponential Implementation
+
+```pseudo
+String toStringAsExponential([int? fractionDigits]);
+```
+
+The `toStringAsExponential` method converts the input number to a `double` before computing the exponential string representation of that number.
+
+If the parameter `fractionDigits` is given, then it must be an integer satisfying:
+
+  0 <= `fractionDigits` <= 20.
+
+Meaning the `fractionDigits` must be a Non-Zero and a Positive value and must be less than 21.
+
+Otherwise, without the parameter, the returned string uses the shortest number of digits that accurately represent this number.
+
+If `fractionDigits` equals 0, then the decimal point is omitted.
+
+## Example
+
+```dart
+void main() {
+
+  var a = 1.toStringAsExponential(3);
+  var b = (256463.51554546).toStringAsExponential(3);
+  var c = (2351.44464661).toStringAsExponential();
+  var d = 254645466111.toStringAsExponential(3);
+  var e = 1224.toStringAsExponential(20);
+  var f = 5.25.toStringAsExponential(0);
+
+  print(a);
+  print(b);
+  print(c);
+  print(d);
+  print(e);
+  print(f);
+
+}
+```
+
+The `Shell` output:
+
+```Shell
+1.000e+0
+2.565e+5
+2.35144464661e+3
+2.546e+11
+1.22400000000000000000e+3
+5e+0
+```
+
+---
```

Comment:
We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

Reply:
Done